src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -115,26 +119,21 @@ public partial class CultureInfo : IFormatProvider, ICloneable
         private static volatile CultureInfo s_DefaultThreadCurrentUICulture;
         private static volatile CultureInfo s_DefaultThreadCurrentCulture;
 
-        [ThreadStatic]
-        private static CultureInfo s_currentThreadCulture;
```

Comment:
We should move there thread-locals to Culture in CoreCLR, so that this can stay in this shared part.

Reply:
That's what I originally wanted to do, but then I realized that there's quite a few dependencies from the C code on this particular implementation. If this is the preferred way then I will refactor that first and make the code sharing patch after.

Reply:
@jkotas I am afraid this refactoring in CoreCLR is way over my head. I started it, but I run into way too many issues and code paths I am not comfortable to modify.

The current implementation is somewhat broken, one can call `CultureInfo.CurrentCulture = something;` and it will never result into call to `nativeInitCultureAccessors`, which is present only in the setter in the the internal `Thread` class. This can become a problem later when some COM call is made with the `IsCultureAware()` code path because it won't get the thread's culture correctly and it will try to restore it to some bogus value. It will call `Thread::GetCulture`, the fast path will fail because of the assumption that `nativeInitCultureAccessors` is called every time a managed code sets `CurrentCulture` and the fallback path then goes straight to the default value instead of calling the property getter.

Moreover the C API assumes that you can get/set culture of thread other than the current one. That is not allowed from managed code in .NET Core. It is certainly possible to implement it, but I am not sure if there's any value in it since it seems that feature is never used and poorly tested.